src/ADS1220.cpp

```diff
@@ -74,6 +74,67 @@ long ADS1220::readADC() {
   return adcVal;
 }
 
+byte * ADS1220::readADC_Array() {
+  digitalWrite(ADS1220_CS_PIN, LOW); // Take CS low
+  delayMicroseconds(1); // Minimum of td(CSSC)
+  static byte dataarray[3];
+  for (int x = 0; x < 3 ; x++)
+  {
+    dataarray[x] = SPI.transfer(SPI_MASTER_DUMMY);
+  }
+  delayMicroseconds(1); // Minimum of td(CSSC)
+  digitalWrite(ADS1220_CS_PIN, HIGH);
+  return dataarray;
+}
+
+//Single Conversion read modes
+long ADS1220::readADC_Single() {
+  digitalWrite(ADS1220_CS_PIN, LOW); // Take CS low
+  delayMicroseconds(1); // Minimum of td(CSSC)
+
+  SPI.transfer(0x08);
+  while(digitalRead(ADS1220_DRDY_PIN) == HIGH)
+  {
+    //Wait to DRDY goes down
+    //Not a good thing
+    //Code could be stuck here
+    //Need a improve later
+  }
+  
+  long adcVal = SPI.transfer(SPI_MASTER_DUMMY);
+  adcVal = (adcVal << 8) | SPI.transfer(SPI_MASTER_DUMMY);
+  adcVal = (adcVal << 8) | SPI.transfer(SPI_MASTER_DUMMY);
+
+  adcVal = ( adcVal << 8 );
+  adcVal = ( adcVal >> 8 );
+  delayMicroseconds(1); // Minimum of td(CSSC)
+  digitalWrite(ADS1220_CS_PIN, HIGH);
+  return adcVal;
+}
+
+byte * ADS1220::readADC_SingleArray() {
+  digitalWrite(ADS1220_CS_PIN, LOW); // Take CS low
+  delayMicroseconds(1); // Minimum of td(CSSC)
+
+  SPI.transfer(0x08);
+  while(digitalRead(ADS1220_DRDY_PIN) == HIGH)
```

Comment:
What about assuming that the user is checking for the DRDY pin state before calling this? If it's high on entry, return zeros? Would prevent things getting stuck here.

Reply:
I will think a workaround about this, but it's been running here since I sent you the code and didn't got any problems.

Reply:
I will think a workaround about this, but it's been running here since I sent you the code and didn't got any problems.

Reply:
It will stay high until he got data to be fecht from the ADC. We can put a timing with milis() to for example, after 100ms if it doesn't go zero, it returns a zero value. Can be a interesting workaround for this.